File: src/main/java/com/nkming/utils/unit/DimensionUtils.java
/*
 * DimensionUtils.java
 *
 * Author: Ming Tsang
 * Copyright (c) 2015 Ming Tsang
 * Refer to LICENSE for details
 */

package com.nkming.utils.unit;

import android.content.Context;
import android.util.DisplayMetrics;
import android.util.TypedValue;

public class DimensionUtils
{
	public static float dpToPx(Context context, float dp)
	{
		DisplayMetrics dm = context.getResources().getDisplayMetrics();
		return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, dm);
	}

	public static float spToPx(Context context, float sp)
	{
		DisplayMetrics dm = context.getResources().getDisplayMetrics();
		return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_SP, sp, dm);
	}

	public static float pxToDp(Context context, float px)
	{
		DisplayMetrics dm = context.getResources().getDisplayMetrics();
		return px * (DisplayMetrics.DENSITY_DEFAULT / (float)dm.densityDpi);
	}
}

File: src/main/kotlin/com/nkming/utils/unit/DimensionUtils.kt
package com.nkming.utils.unit;

import android.content.Context
import android.util.TypedValue

class DimensionUtils
{
	companion object
	{
		@JvmStatic
		fun dpToPx(context: Context, dp: Float): Float
		{
			val dm = context.resources.displayMetrics
			return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, dm)
		}

		@JvmStatic
		fun spToPx(context: Context, sp: Float): Float
		{
			val dm = context.resources.displayMetrics
			return TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_SP, sp, dm)
		}

		@JvmStatic
		fun pxToDp(context: Context, px: Float): Float
		{
			val dm = context.resources.displayMetrics
			return px / dm.density
		}

		@JvmStatic
		fun pxToSp(context: Context, px: Float): Float
		{
			val dm = context.resources.displayMetrics
			return px / dm.scaledDensity
		}
	}
}
